From 9fd06e864d252799c6001df1a61611f41c9dadae Mon Sep 17 00:00:00 2001
From: Rob Herring <r.herring@freescale.com>
Date: Fri, 6 Feb 2009 16:52:58 -0600
Subject: [PATCH 073/505] ENGR00107392-2 Move asm/arch headers to linux

Architecture specific headers moved and are not accessible to userspace in 2.6.28.
So they have been moved to linux include directory.

Update test modules to work on 2.6.28

Signed-off-by: Rob Herring <r.herring@freescale.com>
---
 include/scc2_test_driver.h                         |   5 +-
 include/scc_test_driver.h                          |  14 +-
 module_test/memalloc.c                             |   5 +-
 module_test/mxc_sdma_mem_test.c                    |  11 +-
 test/imx_adc/imx_adc_test.c                        |   2 +-
 .../pmic_testapp_rtc/pmic_rtc_test.h               |   4 +-
 .../protocol_tests/mcu_pmic_protocol_test.c        |   4 +-
 test/mxc_asrc_test/mxc_asrc_test.c                 |   4 +-
 test/mxc_fb_test/mxc_fb_test.c                     |   2 +-
 test/mxc_fb_test/mxc_fb_vsync_test.c               |   4 +-
 test/mxc_ipudev_test/mxc_ipu_dev.c                 |   4 +-
 test/mxc_mlb_test/mxc_mlb_test.c                   |   2 +-
 test/mxc_pf_test/mxc_pf_test.c                     |   2 +-
 .../mxc_pmic_test/pmic_testapp_adc/pmic_adc_test.h |   4 +-
 .../pmic_testapp_battery/pmic_battery_test.h       |   4 +-
 .../pmic_testapp_light/pmic_testapp_light.c        |   4 +-
 .../mxc_pmic_test/pmic_testapp_rtc/pmic_rtc_test.h |   2 +-
 .../protocol_tests/mc13783_test_common.h           |   2 +-
 .../protocol_tests/mc34704_test_common.h           |   4 +-
 test/mxc_scc2/Makefile                             |   2 +
 test/mxc_scc2/fsl_shw.h                            |   2 +-
 test/mxc_scc2/scc2_test_driver.h                   | 207 ---------------------
 test/mxc_v4l2_test/mxc_v4l2_capture.c              |   2 +-
 test/mxc_v4l2_test/mxc_v4l2_output.c               |   6 +-
 test/mxc_v4l2_test/mxc_v4l2_overlay.c              |   2 +-
 test/mxc_v4l2_test/mxc_v4l2_tvin.c                 |   2 +-
 26 files changed, 46 insertions(+), 260 deletions(-)
 delete mode 100644 test/mxc_scc2/scc2_test_driver.h

diff --git a/include/scc2_test_driver.h b/include/scc2_test_driver.h
index 77bfe2b..cf08d51 100644
--- a/include/scc2_test_driver.h
+++ b/include/scc2_test_driver.h
@@ -1,5 +1,5 @@
 /*
- * Copyright 2004-2008 Freescale Semiconductor, Inc. All rights reserved.
+ * Copyright 2004-2009 Freescale Semiconductor, Inc. All rights reserved.
  */
 
 /*
@@ -24,12 +24,11 @@
 #ifdef __KERNEL__
 
 #include <portable_os.h>
-#include <asm/arch/hardware.h>
 
 #endif /* kernel */
 
 
-#include <asm/arch/mxc_scc2_driver.h>
+#include <linux/mxc_scc2_driver.h>
 
 
 #ifndef SCC2_TEST_DRIVER_NAME
diff --git a/include/scc_test_driver.h b/include/scc_test_driver.h
index 4c83c72..a97871d 100644
--- a/include/scc_test_driver.h
+++ b/include/scc_test_driver.h
@@ -1,5 +1,5 @@
 /*
- * Copyright 2004-2006 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2004-2009 Freescale Semiconductor, Inc. All Rights Reserved.
  */
 
 /*
@@ -29,19 +29,9 @@
 
 #include <linux/mm.h>           /* for io_remap_page_range() */
 
-#ifdef TAHITI
-#include <asm/arch/mx2.h>           /* MX21 board memory map*/
-#define SCC_BASE SMN_BASE_ADDR
-#elif defined(MXC)
-#include <asm/arch/hardware.h>
-
-#else
-DO_NOT_KNOW_TARGET_ARCH;
-#endif
-
 #endif /* kernel */
 
-#include <asm/arch/mxc_scc_driver.h>
+#include <linux/mxc_scc_driver.h>
 
 #ifdef __KERNEL__
 
diff --git a/module_test/memalloc.c b/module_test/memalloc.c
index e54d152..b6a7374 100644
--- a/module_test/memalloc.c
+++ b/module_test/memalloc.c
@@ -369,7 +369,10 @@ s32 __init memalloc_init(void)
 		pr_debug("memalloc: error creating memalloc class\n");
 		goto error1;
 	}
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,26))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,28))
+	temp_class = device_create(memalloc_class, NULL,
+				   MKDEV(memalloc_major, 0), NULL, "memalloc");
+#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,26))
 	temp_class = device_create(memalloc_class, NULL,
 				   MKDEV(memalloc_major, 0), "memalloc");
 #else
diff --git a/module_test/mxc_sdma_mem_test.c b/module_test/mxc_sdma_mem_test.c
index 19db884..c2ddc54 100644
--- a/module_test/mxc_sdma_mem_test.c
+++ b/module_test/mxc_sdma_mem_test.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2006-2008 Freescale Semiconductor, Inc. All rights reserved.
+ * Copyright 2006-2009 Freescale Semiconductor, Inc. All rights reserved.
  */
 
 /*
@@ -28,10 +28,6 @@
 #include <linux/io.h>
 #include <linux/delay.h>
 
-#include <asm/arch/dma.h>
-
-#include <asm/arch/hardware.h>
-
 static int gMajor; //major number of device
 static struct class *dma_tm_class;
 static char *wbuf;
@@ -234,7 +230,10 @@ int __init sdma_init_module(void)
 		return PTR_ERR(dma_tm_class);
 	}
 
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,26))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,28))
+	temp_class = device_create(dma_tm_class, NULL,
+				   MKDEV(gMajor, 0), NULL, "sdma_test");
+#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,26))
 	temp_class = device_create(dma_tm_class, NULL,
 				   MKDEV(gMajor, 0), "sdma_test");
 #else
diff --git a/test/imx_adc/imx_adc_test.c b/test/imx_adc/imx_adc_test.c
index c503643..674d015 100644
--- a/test/imx_adc/imx_adc_test.c
+++ b/test/imx_adc/imx_adc_test.c
@@ -20,7 +20,7 @@
 #include <pthread.h>
 #include <fcntl.h>
 #include <linux/input.h>
-#include <asm/arch/imx_adc.h>
+#include <linux/imx_adc.h>
 
 int main(void)
 {
diff --git a/test/max8660_pmic_test/pmic_testapp_rtc/pmic_rtc_test.h b/test/max8660_pmic_test/pmic_testapp_rtc/pmic_rtc_test.h
index eb0d0fd..62c465d 100644
--- a/test/max8660_pmic_test/pmic_testapp_rtc/pmic_rtc_test.h
+++ b/test/max8660_pmic_test/pmic_testapp_rtc/pmic_rtc_test.h
@@ -1,5 +1,5 @@
 /*
- * Copyright 2008 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2008-2009 Freescale Semiconductor, Inc. All Rights Reserved.
  */
 
 /*
@@ -37,7 +37,7 @@ extern "C"{
 #include <errno.h>
 #include <linux/wait.h>
 
-#include <asm/arch/pmic_rtc.h>
+#include <linux/pmic_rtc.h>
 
 /*==============================================================================
                                        DEFINES AND MACROS
diff --git a/test/max8660_pmic_test/protocol_tests/mcu_pmic_protocol_test.c b/test/max8660_pmic_test/protocol_tests/mcu_pmic_protocol_test.c
index 821efb0..a82d81e 100644
--- a/test/max8660_pmic_test/protocol_tests/mcu_pmic_protocol_test.c
+++ b/test/max8660_pmic_test/protocol_tests/mcu_pmic_protocol_test.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2008 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2008-2009 Freescale Semiconductor, Inc. All Rights Reserved.
  */
 
 /*
@@ -25,7 +25,7 @@
 #include <fcntl.h>
 
 #include <errno.h>
-#include <asm/arch/pmic_external.h>
+#include <linux/pmic_external.h>
 
 #define TFAIL -1
 #define TPASS 0
diff --git a/test/mxc_asrc_test/mxc_asrc_test.c b/test/mxc_asrc_test/mxc_asrc_test.c
index 3e0c781..3d646c1 100644
--- a/test/mxc_asrc_test/mxc_asrc_test.c
+++ b/test/mxc_asrc_test/mxc_asrc_test.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2008 Freescale Semiconductor, Inc. All rights reserved.
+ * Copyright 2008-2009 Freescale Semiconductor, Inc. All rights reserved.
  */
 
 /*
@@ -27,7 +27,7 @@
 #include <string.h>
 #include <malloc.h>
 #include <sys/time.h>
-#include <asm/arch/mxc_asrc.h>
+#include <linux/mxc_asrc.h>
 
 #define DMA_BUF_SIZE 4096
 #define BUF_NUM 4
diff --git a/test/mxc_fb_test/mxc_fb_test.c b/test/mxc_fb_test/mxc_fb_test.c
index a36e921..d243ef3 100644
--- a/test/mxc_fb_test/mxc_fb_test.c
+++ b/test/mxc_fb_test/mxc_fb_test.c
@@ -34,7 +34,7 @@ extern "C"{
 #include <fcntl.h>
 #include <sys/ioctl.h>
 #include <unistd.h>
-#include <asm/arch/mxcfb.h>
+#include <linux/mxcfb.h>
 #include <sys/mman.h>
 #include <math.h>
 #include <string.h>
diff --git a/test/mxc_fb_test/mxc_fb_vsync_test.c b/test/mxc_fb_test/mxc_fb_vsync_test.c
index cbfa3c0..5de7205 100644
--- a/test/mxc_fb_test/mxc_fb_vsync_test.c
+++ b/test/mxc_fb_test/mxc_fb_vsync_test.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2004-2008 Freescale Semiconductor, Inc. All rights reserved.
+ * Copyright 2004-2009 Freescale Semiconductor, Inc. All rights reserved.
  */
 
 /*
@@ -29,7 +29,7 @@
 #include <math.h>
 #include <string.h>
 
-#include <asm/arch/mxcfb.h>
+#include <linux/mxcfb.h>
 
 
 int fd_fb = 0;
diff --git a/test/mxc_ipudev_test/mxc_ipu_dev.c b/test/mxc_ipudev_test/mxc_ipu_dev.c
index ea2bc3a..d8ad8be 100644
--- a/test/mxc_ipudev_test/mxc_ipu_dev.c
+++ b/test/mxc_ipudev_test/mxc_ipu_dev.c
@@ -33,8 +33,8 @@
 #include <sys/mman.h>
 #include <linux/videodev.h>
 #include <linux/autoconf.h>
-#include <asm/arch/ipu.h>
-#include <asm/arch/mxcfb.h>
+#include <linux/ipu.h>
+#include <linux/mxcfb.h>
 
 #ifdef CONFIG_MXC_IPU_V1
 #define ROT_BEGIN	IPU_ROTATE_90_RIGHT
diff --git a/test/mxc_mlb_test/mxc_mlb_test.c b/test/mxc_mlb_test/mxc_mlb_test.c
index f480f42..70c9c6d 100644
--- a/test/mxc_mlb_test/mxc_mlb_test.c
+++ b/test/mxc_mlb_test/mxc_mlb_test.c
@@ -25,7 +25,7 @@
 #include <errno.h>
 #include <string.h>
 
-#include <asm/arch/mxc_mlb.h>
+#include <linux/mxc_mlb.h>
 
 #define MLB_ASYNC_DEV	"/dev/async"
 #define MLB_CTRL_DEV	"/dev/ctrl"
diff --git a/test/mxc_pf_test/mxc_pf_test.c b/test/mxc_pf_test/mxc_pf_test.c
index 3870426..d350283 100644
--- a/test/mxc_pf_test/mxc_pf_test.c
+++ b/test/mxc_pf_test/mxc_pf_test.c
@@ -24,7 +24,7 @@
 #include <sys/resource.h>
 #include <linux/types.h>
 
-#include <asm/arch/mxc_pf.h>
+#include <linux/mxc_pf.h>
 
 
 
diff --git a/test/mxc_pmic_test/pmic_testapp_adc/pmic_adc_test.h b/test/mxc_pmic_test/pmic_testapp_adc/pmic_adc_test.h
index 09e2411..cfd5803 100644
--- a/test/mxc_pmic_test/pmic_testapp_adc/pmic_adc_test.h
+++ b/test/mxc_pmic_test/pmic_testapp_adc/pmic_adc_test.h
@@ -35,8 +35,8 @@ extern "C"{
 #include <stdio.h>	/* sscanf() & perror() */
 #include <stdlib.h>	/* atoi() */
 
-#include <asm/arch/pmic_external.h>
-#include <asm/arch/pmic_adc.h>
+#include <linux/pmic_external.h>
+#include <linux/pmic_adc.h>
 
 /*==============================================================================
                                            CONSTANTS
diff --git a/test/mxc_pmic_test/pmic_testapp_battery/pmic_battery_test.h b/test/mxc_pmic_test/pmic_testapp_battery/pmic_battery_test.h
index 54f2131..96a7f4d 100644
--- a/test/mxc_pmic_test/pmic_testapp_battery/pmic_battery_test.h
+++ b/test/mxc_pmic_test/pmic_testapp_battery/pmic_battery_test.h
@@ -37,8 +37,8 @@ extern "C"{
 #include <errno.h>
 #include <linux/wait.h>
 
-#include <asm/arch/pmic_status.h>
-#include <asm/arch/pmic_battery.h>
+#include <linux/pmic_status.h>
+#include <linux/pmic_battery.h>
 
 /*==============================================================================
                                        DEFINES AND MACROS
diff --git a/test/mxc_pmic_test/pmic_testapp_light/pmic_testapp_light.c b/test/mxc_pmic_test/pmic_testapp_light/pmic_testapp_light.c
index e0416c3..854b571 100644
--- a/test/mxc_pmic_test/pmic_testapp_light/pmic_testapp_light.c
+++ b/test/mxc_pmic_test/pmic_testapp_light/pmic_testapp_light.c
@@ -23,8 +23,8 @@
 #include <stdio.h>		/* sscanf() & perror() */
 #include <stdlib.h>		/* atoi() */
 
-#include <asm/arch/pmic_external.h>
-#include <asm/arch/pmic_light.h>
+#include <linux/pmic_external.h>
+#include <linux/pmic_light.h>
 
 #include "../include/test.h"
 #include "../include/usctest.h"
diff --git a/test/mxc_pmic_test/pmic_testapp_rtc/pmic_rtc_test.h b/test/mxc_pmic_test/pmic_testapp_rtc/pmic_rtc_test.h
index f70bc2d..cb097ea 100644
--- a/test/mxc_pmic_test/pmic_testapp_rtc/pmic_rtc_test.h
+++ b/test/mxc_pmic_test/pmic_testapp_rtc/pmic_rtc_test.h
@@ -37,7 +37,7 @@ extern "C"{
 #include <errno.h>
 #include <linux/wait.h>
 
-#include <asm/arch/pmic_rtc.h>
+#include <linux/pmic_rtc.h>
 
 /*==============================================================================
                                        DEFINES AND MACROS
diff --git a/test/mxc_pmic_test/protocol_tests/mc13783_test_common.h b/test/mxc_pmic_test/protocol_tests/mc13783_test_common.h
index 4dde4a8..2015857 100644
--- a/test/mxc_pmic_test/protocol_tests/mc13783_test_common.h
+++ b/test/mxc_pmic_test/protocol_tests/mc13783_test_common.h
@@ -36,7 +36,7 @@ extern "C" {
 #include <stdlib.h>		/* atoi() */
 #include <pthread.h>
 
-#include <asm/arch/pmic_external.h>
+#include <linux/pmic_external.h>
 
 /*==============================================================================
                                            CONSTANTS
diff --git a/test/mxc_pmic_test/protocol_tests/mc34704_test_common.h b/test/mxc_pmic_test/protocol_tests/mc34704_test_common.h
index a3ba856..ddc4542 100644
--- a/test/mxc_pmic_test/protocol_tests/mc34704_test_common.h
+++ b/test/mxc_pmic_test/protocol_tests/mc34704_test_common.h
@@ -1,5 +1,5 @@
 /*
- * Copyright 2008 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2008-2009 Freescale Semiconductor, Inc. All Rights Reserved.
  */
 
 /*
@@ -36,7 +36,7 @@ extern "C" {
 #include <stdlib.h>		/* atoi() */
 #include <pthread.h>
 
-#include <asm/arch/pmic_external.h>
+#include <linux/pmic_external.h>
 
 /*==============================================================================
                                            CONSTANTS
diff --git a/test/mxc_scc2/Makefile b/test/mxc_scc2/Makefile
index 196d84f..85e8f04 100644
--- a/test/mxc_scc2/Makefile
+++ b/test/mxc_scc2/Makefile
@@ -7,6 +7,8 @@ else
 OBJS =
 endif
 
+CFLAGS += -DFSL_PLATFORM_OTHER -DFSL_HAVE_SCC2
+
 all : $(OBJS)
 	cp -r *.sh $(OBJDIR)
 
diff --git a/test/mxc_scc2/fsl_shw.h b/test/mxc_scc2/fsl_shw.h
index 8f0159b..6f224c4 100644
--- a/test/mxc_scc2/fsl_shw.h
+++ b/test/mxc_scc2/fsl_shw.h
@@ -294,7 +294,7 @@
 
 #else
 
-#error FSL_SHW_API_platform_not_recognized
+#include "shw_driver.h"
 
 #endif
 
diff --git a/test/mxc_scc2/scc2_test_driver.h b/test/mxc_scc2/scc2_test_driver.h
deleted file mode 100644
index 4089aeb..0000000
--- a/test/mxc_scc2/scc2_test_driver.h
+++ /dev/null
@@ -1,207 +0,0 @@
-/*
- * Copyright 2004-2008 Freescale Semiconductor, Inc. All rights reserved.
- */
-
-/*
- * The code contained herein is licensed under the GNU General Public
- * License. You may obtain a copy of the GNU General Public License
- * Version 2 or later at the following locations:
- *
- * http://www.opensource.org/licenses/gpl-license.html
- * http://www.gnu.org/copyleft/gpl.html
- */
-
-/*! @file scc2_test_driver.h
- *
- *  This header file provides definitions needed by the SCC2 test driver.
- *
- */
-
-
-#ifndef SCC2_TEST_DRIVER_H
-#define SCC2_TEST_DRIVER_H
-
-#ifdef __KERNEL__
-
-#include <portable_os.h>
-#include <asm/arch/hardware.h>
-
-#endif /* kernel */
-
-#include <asm/arch/mxc_scc2_driver.h>
-
-#ifndef SCC2_TEST_DRIVER_NAME
-
-/** /dev/xxx name for this device */
-#define SCC2_TEST_DRIVER_NAME "scc2_test"
-
-
-#endif /* kernel */
-
-
-/* User/Driver interface definitions */
-
-/**
- * @defgroup ioctlreturns Ioctl Error Codes.
- *
- * These are the values returned by #scc_ioctl and placed into @c
- * errno by @c ioctl. Porting opportunity.  These values were chosen to
- * match standard Linux values.
- */
-/** @addtogroup ioctlreturns */
-/** @{ */
-#define IOCTL_SCC2_OK             0      /**< @c ioctl completed successfully */
-#define IOCTL_SCC2_INVALID_CMD    ENOTTY /**< Invalid command passed. */
-#define IOCTL_SCC2_INVALID_MODE   EINVAL /**< Invalid cipher mode. */
-#define IOCTL_SCC2_SCM_BUSY       EBUSY  /**< SCM is busy */
-#define IOCTL_SCC2_IMPROPER_STATE EBADFD /**< Improper state for operation */
-#define IOCTL_SCC2_IMPROPER_ADDR  EFAULT /**< Improper address/offset passed */
-#define IOCTL_SCC2_NO_MEMORY      ENOMEM /**< Insufficient memory to process */
-#define IOCTL_SCC2_FAILURE        ESPIPE /**< Generic 'SCC2 error' error */
-/** @} */
-
-/*
- * Interface definitions between user and driver
- */
-
-/* This is a porting opportunity.  It identifies the 'unique' byte
-   value inserted into the IOCTL number.  It really only has to be
-   unique within the software used on a given device. */
-#ifndef SCC2_TEST_DRIVER_IOCTL_IDENTIFIER
-#define SCC2_TEST_DRIVER_IOCTL_IDENTIFIER 's'
-#endif
-
-/* Define SCC2 Driver Commands (Argument 2 of ioctl) */
-
-/** @defgroup ioctlcmds ioctl command (argument 2) values */
-/** @addtogroup ioctlcmds */
-/** @{ */
-/** ioctl cmd to return version and configuration information on driver and
- *  SCC2 */
-#ifdef _IOW
-#undef _IOW
-#endif
-#ifdef _IOWR
-#undef _IOWR
-#endif
-#ifdef _IO
-#undef _IO
-#endif
-
-#define _IOW(x, y, z) y
-#define _IOWR(x, y, z) y
-#define _IO(x, y) y
-
-#define SCC2_TEST_GET_CONFIGURATION _IOW (SCC2_TEST_DRIVER_IOCTL_IDENTIFIER, 0, \
-         scc_configuration_access)
-/** ioctl cmd to test the scc_read_register() function of the SCC2 driver */
-#define SCC2_TEST_READ_REG         _IOWR(SCC2_TEST_DRIVER_IOCTL_IDENTIFIER, 1, \
-         scc_reg_access)
-/** ioctl cmd to test the scc_write_register() function of the SCC2 driver */
-#define SCC2_TEST_WRITE_REG        _IOWR(SCC2_TEST_DRIVER_IOCTL_IDENTIFIER, 2, \
-         scc_reg_access)
-/** ioctl cmd to test the scc_crypt() function of the SCC2 driver */
-#define SCC2_TEST_ENCRYPT          _IOWR(SCC2_TEST_DRIVER_IOCTL_IDENTIFIER, 3, \
-         scc_encrypt_decrypt)
-/** ioctl cmd to test the scc_crypt() function of the SCC2 driver */
-#define SCC2_TEST_DECRYPT          _IOWR(SCC2_TEST_DRIVER_IOCTL_IDENTIFIER, 4, \
-         scc_encrypt_decrypt)
-/** ioctl cmd to test the scc_set_sw_alarm() function of the SCC2 driver */
-#define SCC2_TEST_SET_ALARM        _IO  (SCC2_TEST_DRIVER_IOCTL_IDENTIFIER, 5)
-/** ioctl cmd to test the scc_zeroize_memories() function of the SCC2 driver */
-#define SCC2_TEST_ZEROIZE          _IOWR(SCC2_TEST_DRIVER_IOCTL_IDENTIFIER, 6, \
-         scc_return_t)
-/** ioctl cmd to test partition allocation function of the SCC2 driver */
-#define SCC2_TEST_ALLOC_PART       _IOWR(SCC2_TEST_DRIVER_IOCTL_IDENTIFIER, 12, \
-        scc_partition_access)
-/** ioctl cmd to test partition engagement function of the SCC2 driver */
-#define SCC2_TEST_ENGAGE_PART      _IOWR(SCC2_TEST_DRIVER_IOCTL_IDENTIFIER, 13, \
-        scc_partition_access)
-/** ioctl cmd to test partition cipher function of the SCC2 driver */
-#define SCC2_TEST_ENCRYPT_PART     _IOWR(SCC2_TEST_DRIVER_IOCTL_IDENTIFIER, 14, \
-        scc_part_cipher_access)
-#define SCC2_TEST_DECRYPT_PART     _IOWR(SCC2_TEST_DRIVER_IOCTL_IDENTIFIER, 15, \
-        scc_part_cipher_access)
-/** ioctl cmd to test partition allocation function of the SCC2 driver */
-#define SCC2_TEST_RELEASE_PART     _IOWR(SCC2_TEST_DRIVER_IOCTL_IDENTIFIER, 16, \
-        scc_partition_access)
-/** ioctl cmd to load data into an SCC2 partition */
-#define SCC2_TEST_LOAD_PART        _IOWR(SCC2_TEST_DRIVER_IOCTL_IDENTIFIER, 17, \
-        scc_part_cipher_access)
-/** ioctl cmd to lread data from an SCC2 partition */
-#define SCC2_TEST_READ_PART        _IOWR(SCC2_TEST_DRIVER_IOCTL_IDENTIFIER, 18, \
-        scc_part_cipher_access)
-
-/** @} */
-
-
-/** @defgroup ioctlStructs Special structs for argument 3 of ioctl
- *
- *
- */
-/** @addtogroup ioctlStructs */
-/** @{ */
-
-/**
- * ioctl structure for retrieving driver & SCC2 version ids, used
- * with #SCC2_TEST_GET_CONFIGURATION.
- */
-typedef struct {
-    int driver_major_version;   /**< Major version of the SCC2 driver code  */
-    int driver_minor_version;   /**< Minor version of the SCC2 driver code  */
-    int scm_version;            /**< from Configuration register */
-    int smn_version;            /**< from SMN Status register */
-    int block_size;             /**< bytes in a block */
-    int black_ram_size;         /**< number of blocks of Black RAM */
-    int red_ram_size;           /**< number of blocks of Red RAM */
-} scc_configuration_access;
-
-
-/**
- * ioctl structure for accessing SCC2 registers, used with
- * #SCC2_TEST_READ_REG and #SCC2_TEST_WRITE_REG.
- */
-typedef struct {
-    uint32_t reg_offset;        /**< The register address from Memory Map */
-    uint32_t reg_data;          /**< Data to/from the register */
-    scc_return_t      function_return_code; /**< Straight from SCC2 driver */
-} scc_reg_access;
-
-
-/**
- * ioctl structure for partition setup/teardown functions
- */
-typedef struct {
-    uint32_t virt_address;      /**< Its virtual address */
-    union {
-        /* for allocate */
-        struct {
-            uint32_t smid;      /**< From user  */
-            uint32_t part_no;   /**< Which partition was allocated */
-            uint32_t phys_address; /**<Its physical address */
-        };
-        /* for engage */
-        struct {
-            const uint8_t* umid; /**< 16-byte UMID for partition, or NULL */
-            uint32_t permissions; /**< Mode register setting for initial access  */
-        };
-    };
-    scc_return_t scc_status;    /**< Returned status of call to scc2 driver */
-} scc_partition_access;
-
-/** @} */
-
-/**
- * ioctl structure for partition encrypt/decrypt functions
- */
-typedef struct {
-    uint32_t virt_address;      /**< Its virtual address */
-    uint32_t red_offset;        /**< Byte offset into partition */
-    uint8_t* black_address;     /**< Virtual address of Black data */
-    uint32_t size_bytes;        /**< Number of bytes of red/black data */
-    uint64_t iv;                /**< first half of IV for CBC mode; second half
-                                   is zero */
-    scc_return_t scc_status;    /**< Returned status of call to scc2 driver */
-} scc_part_cipher_access;
-
-#endif /* scc2_test_driver.h */
diff --git a/test/mxc_v4l2_test/mxc_v4l2_capture.c b/test/mxc_v4l2_test/mxc_v4l2_capture.c
index 4c99c63..78ad5f3 100644
--- a/test/mxc_v4l2_test/mxc_v4l2_capture.c
+++ b/test/mxc_v4l2_test/mxc_v4l2_capture.c
@@ -43,7 +43,7 @@ extern "C"{
 #include <sys/mman.h>
 #include <string.h>
 #include <malloc.h>
-//#include <asm/arch/mxc_v4l2.h>
+//#include <linux/mxc_v4l2.h>
 
 struct v4l2_mxc_offset {
 	uint32_t u_offset;
diff --git a/test/mxc_v4l2_test/mxc_v4l2_output.c b/test/mxc_v4l2_test/mxc_v4l2_output.c
index 8a91e57..d7a9fda 100644
--- a/test/mxc_v4l2_test/mxc_v4l2_output.c
+++ b/test/mxc_v4l2_test/mxc_v4l2_output.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2004-2006 Freescale Semiconductor, Inc. All rights reserved.
+ * Copyright 2004-2009 Freescale Semiconductor, Inc. All rights reserved.
  */
 
 /*
@@ -42,8 +42,8 @@ extern "C"{
 #include <malloc.h>
 #include <sys/time.h>
 
-#include <asm/arch/mxcfb.h>
-//#include <asm/arch/mxc_v4l2.h>
+#include <linux/mxcfb.h>
+//#include <linux/mxc_v4l2.h>
 
 struct v4l2_mxc_offset {
 	uint32_t u_offset;
diff --git a/test/mxc_v4l2_test/mxc_v4l2_overlay.c b/test/mxc_v4l2_test/mxc_v4l2_overlay.c
index 13b1579..433ff3b 100644
--- a/test/mxc_v4l2_test/mxc_v4l2_overlay.c
+++ b/test/mxc_v4l2_test/mxc_v4l2_overlay.c
@@ -47,7 +47,7 @@ extern "C"{
 #include <string.h>
 #include <malloc.h>
 
-#include <asm/arch/mxcfb.h>
+#include <linux/mxcfb.h>
 
 #define TFAIL -1
 #define TPASS 0
diff --git a/test/mxc_v4l2_test/mxc_v4l2_tvin.c b/test/mxc_v4l2_test/mxc_v4l2_tvin.c
index e5c98c0..b78c489 100644
--- a/test/mxc_v4l2_test/mxc_v4l2_tvin.c
+++ b/test/mxc_v4l2_test/mxc_v4l2_tvin.c
@@ -48,7 +48,7 @@ extern "C"{
 #include <string.h>
 #include <malloc.h>
 
-#include <asm/arch/mxcfb.h>
+#include <linux/mxcfb.h>
 
 #define TFAIL -1
 #define TPASS 0
-- 
1.8.0

